### DISCUSSION TITLE
Renovate should allow ed25519 keys for gitPrivateKey

### DISCUSSION THREAD

**Discussion by wwuck** [renovatebot/renovate#12558] [Suggest an Idea]
### What would you like Renovate to be able to do?

It would be nice to have renovate allow ed25519 keys for gitPrivateKey config.

### If you have any ideas on how this should be implemented, please tell us here.

I have no idea how ed25519 can be added to the existing support for RSA keys.

### Is this a feature you are interested in implementing yourself?

No

**Comment by viceice** [comment]
Have you tried to simply assign an  `ed25519` key? What errors are happen? I think it should just work.

**Comment by wwuck** [comment]
Hmmm I think I ran into the same problem as https://github.com/renovatebot/renovate/discussions/10017.

Can we get the docs for `gitPrivateKey` updated to mention that the generated PGP/GPG key should not have a passphrase?  Or alternatively, could a new `gitPrivateKeyPassphrase` option be added to allow keys with a passphrase?  Currently, if a key is created with a passphrase and assigned to `gitPrivateKey` config then the resulting error from renovate is confusing and provides no indication of what has gone wrong and how it should be rectified.

**Comment by rarkins** [comment]
I've converted this to a discussion. I'm open to the idea of `gitPrivateKeyPassphrase` but please create a new feature request specifically for that and include an example key with passphrase you've generated (a throwaway one, obviously).

**Comment by wwuck** [comment]
https://github.com/renovatebot/renovate/issues/12566
